### DISCUSSION TITLE
SharePoint Online Site Script - createSPList - Is the addFile SubAction Supported Today in SharePoint Online?

### DISCUSSION THREAD

**Discussion by wniemiGHUb** [pnp/powershell#3031] [Q&A]
We are building a SharePoint Online Site Provisioning process.  We need to know when creating site scripts and scripting the creation of document libraries using the createSPList action, if the addFile subaction is supported today?  We do know that the addFile subaction is listed within the official Microsoft Script Actions Schema found at:  https://github.com/microsoft/json-schemas/blob/main/sp/site-design-script-actions.schema.json.  Is the addFile subaction supported Today in SharePoint Online?

**Comment by MatousAc** [comment]
I'd absolutely love to know this too. I need to use the `addFolder` subaction that's defined, but 1.) I'm not sure if it needs additional parameters, 2. requests to create scripts that contain this never seem to go through.
